internal/server/singleprocess/service_job.go

```diff
@@ -180,6 +228,136 @@ func (s *service) QueueJob(
 	return &pb.QueueJobResponse{JobId: job.Id}, nil
 }
 
+func (s *service) launchOndemandRunner(
+	ctx context.Context,
+	source *pb.Job,
+	project *pb.Project,
+	odref *pb.Ref_OndemandRunner,
+) (string, error) {
+	log := hclog.FromContext(ctx)
+
+	od, err := s.state.OndemandRunnerGet(odref)
+	if err != nil {
+		return "", err
+	}
+
+	runnerId, err := server.Id()
+	if err != nil {
+		return "", err
+	}
+
+	log.Info("requesting ondemand runner via task start", "runner-id", runnerId)
+
+	// Get our server config
+	cfg, err := s.state.ServerConfigGet()
+	if err != nil {
+		return "", err
+	}
+
+	// Follow the same logic as RunnerGetDeploymentConfig, which includes this note:
+	// Our addr for now is just the first one since we don't support
+	// multiple addresses yet. In the future we will want to support more
+	// advanced choicing.
+
+	var addr *pb.ServerConfig_AdvertiseAddr
+
+	// This should only happen during tests
+	if len(cfg.AdvertiseAddrs) == 0 {
+		log.Info("server has no advertise addrs, using localhost")
+		addr = &pb.ServerConfig_AdvertiseAddr{
+			Addr: "localhost:9701",
+		}
+	} else {
+		addr = cfg.AdvertiseAddrs[0]
+	}
+
+	// We generate a new login token for each ondemand-runner used. This will inherit
+	// the user of the token to be the user that queue'd the original job, which is
+	// the correct behavior.
+	token, err := s.newToken(60*time.Minute, DefaultKeyId, nil, &pb.Token{
+		Kind: &pb.Token_Login_{Login: &pb.Token_Login{
+			UserId: DefaultUserId,
+		}},
+	})
+	if err != nil {
+		return "", err
+	}
+
+	envVars := map[string]string{
+		"WAYPOINT_RUNNER_ID":        runnerId,
+		serverclient.EnvServerAddr:  addr.Addr,
+		serverclient.EnvServerToken: token,
+	}
+
+	if addr.Tls {
+		envVars[serverclient.EnvServerTls] = "1"
+	}
+
+	if addr.TlsSkipVerify {
+		envVars[serverclient.EnvServerTlsSkipVerify] = "1"
+	}
+
+	for k, v := range od.EnvironmentVariables {
+		envVars[k] = v
+	}
+
+	args := []string{"runner", "agent", "-vv", "-id", runnerId, "-odr"}
+
+	job := &pb.Job{
+		Workspace:   source.Workspace,
+		Application: source.Application,
+		Operation: &pb.Job_StartTask{
+			StartTask: &pb.Job_StartTaskLaunchOp{
+				Params: &pb.Job_TaskPluginParams{
+					PluginType: od.PluginType,
+					HclConfig:  od.PluginConfig,
+					HclFormat:  od.ConfigFormat,
+				},
+				Info: &pb.TaskLaunchInfo{
+					OciUrl:               od.OciUrl,
+					EnvironmentVariables: envVars,
+					Arguments:            args,
+				},
+			},
+		},
+	}
+
+	// Get the next id
+	id, err := server.Id()
+	if err != nil {
+		return "", status.Errorf(codes.Internal, "uuid generation failed: %s", err)
+	}
+	job.Id = id
+
+	// Validate expiry if we have one
+	job.ExpireTime = nil
+	dur, err := time.ParseDuration("60s")
```

Comment:
This is hardcoded at 60s, is that what you expected? (The comment makes it sound like its some sort of option)